model-optimizer/extensions/middle/ReverseV2ToReverseSequence.py

```diff
@@ -32,29 +35,51 @@ def pattern():
 
     def replace_pattern(self, graph: Graph, match: dict):
         reverse = match['reverse']
-        input_data_shape = reverse.in_node(0).shape
-
-        if len(input_data_shape) == 1:
-            raise Error('Reverse operation name = {} is\'t supported because of 1D input.'.format(reverse.name))
+        reverse_name = reverse.soft_get('name', reverse.id)
 
         assert reverse.in_port(1).disconnected()
 
-        seq_axis = reverse['axis']
-        # We need to choose arbitrary batch_axis != sequence_axis
-        batch_axis = int(not seq_axis)
+        in_shape_rank = len(reverse.in_port(0).data.get_shape())
+        # add new dimension as batch for rank = 1
+        if in_shape_rank == 1:
+            unsq_node = create_op_node_with_second_input(graph, Unsqueeze, int64_array([0]),
+                                                         {'name': reverse_name+"/Unsqueeze"})
+            reverse.in_port(0).get_source().connect(unsq_node.in_port(0))
+            new_in = unsq_node.out_port(0)
+            batch_axis = 0
+            # add 1 for newly added dimension
+            seq_axis = reverse['axis'] + 1 if reverse['axis'] >= 0 else reverse['axis']
+        else:
+            new_in = reverse.in_port(0).get_source()
+            seq_axis = reverse['axis']
+            batch_axis = 0 if seq_axis != 0 else 1
 
         # 1. For ReverseSequence 1-port input is seq_lengths => create this input node
-        seq_lengths = np.ones(input_data_shape[batch_axis]) * input_data_shape[seq_axis]
-
         reverse_name = reverse.soft_get('name',  reverse.id)
         rename_node(reverse, reverse_name + '/to_delete')
```

Comment:
please rename nodes in the end once you did transformation using `rename_nodes([...])`

Reply:
then we will have 2 nodes with the same name in graph until rename - I don't like this idea. Moved removing node at the end